datafusion/functions-aggregate/src/max_by.rs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::{Array, ArrayRef};
+use arrow_schema::DataType;
+use datafusion_common::utils::get_row_at_idx;
+use datafusion_common::{Result, ScalarValue};
+use datafusion_expr::{Accumulator, AggregateUDFImpl, Signature, Volatility};
+use datafusion_functions_aggregate_common::accumulator::AccumulatorArgs;
+use std::any::Any;
+use std::fmt::Debug;
+use std::ops::Deref;
+
+use crate::first_last::FirstValueAccumulator;
+
+make_udaf_expr_and_func!(
+    MaxByFunction,
+    max_by,
+    x y,
+    "Returns the value of the first column corresponding to the maximum value in the second column.",
+    max_by_udaf
+);
+
+pub struct MaxByFunction {
```

Comment:
Instead of adding another function, could we rewrite `max_by(a, b)` to `first(a) order by b desc` or `last(a) order by b`?

Reply:
Yes, I think that would be optimal, I'll modify it tomorrow